![inu_neko_profile_image](inu_neko_profile_image_small.png)


Hello! We are Inu + Neku and we are a Dog & Cat services and supplies store located in New York City. We just started our e-commerce business and need your help analyzing our data!

## 1.1 Description

We need to make sure the data is clean before starting our analysis. As a reminder, we should check for:

* Duplicate records
* Consistente formatting
* Missing values
* Obviously wrong values

We will start by importing all our libraries for analysis and do some exploratory analyis on our data:

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df_cleaned = pd.read_csv('inu_neko_orderline_clean.csv')
df_cleaned

trans_id      prod_upc  cust_id             trans_timestamp  \
0      10300097  719638485153  1001019  2021-01-01 07:35:21.439873   
1      10300093   73201504044  1001015  2021-01-01 09:33:37.499660   
2      10300093  719638485153  1001015  2021-01-01 09:33:37.499660   
3      10300093  441530839394  1001015  2021-01-01 09:33:37.499660   
4      10300093  733426809698  1001015  2021-01-01 09:33:37.499660   
...         ...           ...      ...                         ...   
38218  10327860  287663658863  1022098  2021-06-30 15:37:12.821020   
38219  10327960  140160459467  1022157  2021-06-30 15:45:09.872732   
38220  10328009  425361189561  1022189  2021-06-30 15:57:44.295104   
38221  10328089  733426809698  1022236  2021-06-30 15:59:29.801593   
38222  10328109  717036112695  1011924  2021-06-30 17:30:52.205912   

       trans_year  trans_month  trans_day  trans_hour  trans_quantity  \
0            2021            1          1           1               1   
1            2021            1          1           1               1   
2            2021            1          1           1               1   
3            2021            1          1           1               2   
4            2021            1          1           1               1   
...           ...          ...        ...         ...             ...   
38218        2021            6         30          30               1   
38219        2021            6         30          30               2   
38220        2021            6         30          30               2   
38221        2021            6         30          30               1   
38222        2021            6         30          30               1   

       cust_age    cust_state  prod_price          prod_title prod_category  \
0            20      New York       72.99            Cat Cave       bedding   
1            34      New York       18.95      Purrfect Puree         treat   
2            34      New York       72.99            Cat Cave       bedding   
3            34      New York       28.45     Ball and String           toy   
4            34      New York       18.95       Yum Fish-Dish          food   
...         ...           ...         ...                 ...           ...   
38218        25      New York        9.95  All Veggie Yummies         treat   
38219        31  Pennsylvania       48.95  Snoozer Essentails       bedding   
38220        53    New Jersey       15.99       Snack-em Fish         treat   
38221        23     Tennessee       18.95       Yum Fish-Dish          food   
38222        24  Pennsylvania       60.99         Reddy Beddy       bedding   

      prod_animal_type  total_sales  
0                  cat        72.99  
1                  cat        18.95  
2                  cat        72.99  
3                  cat        56.90  
4                  cat        18.95  
...                ...          ...  
38218              dog         9.95  
38219              dog        97.90  
38220              cat        31.98  
38221              cat        18.95  
38222              dog        60.99  

[38223 rows x 16 columns]

In order to get a better understanding of the about the structure of our data including number of columns, non-null counts, and data type we use the `.info()` method of our pandas library.

In [3]:
df_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38223 entries, 0 to 38222
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   trans_id          38223 non-null  int64  
 1   prod_upc          38223 non-null  int64  
 2   cust_id           38223 non-null  int64  
 3   trans_timestamp   38223 non-null  object 
 4   trans_year        38223 non-null  int64  
 5   trans_month       38223 non-null  int64  
 6   trans_day         38223 non-null  int64  
 7   trans_hour        38223 non-null  int64  
 8   trans_quantity    38223 non-null  int64  
 9   cust_age          38223 non-null  int64  
 10  cust_state        38223 non-null  object 
 11  prod_price        38223 non-null  float64
 12  prod_title        38223 non-null  object 
 13  prod_category     38223 non-null  object 
 14  prod_animal_type  38223 non-null  object 
 15  total_sales       38223 non-null  float64
dtypes: float64(2), int64(9), object(5)
memor

We will now begin the process of exploring our data to answer relevant business questions. A couple of questions come to mind when exploring our data. It is always helpful to have objectives of what we are trying to achieve before starting our exploratory analysis. We came up we 3 main objectives:

1. To understand the top 5 niche pet products to add to our product line by the end of December
2. To identify the top 5 distinguishing demographics of our new customers.
3. To determine the growth/decline of sales in the next 6 months and what factors contribute to it


There are a few business questions that we came up with to help us answer answer achieve our analysis objectives. They include:

1. How many transactions are there
2. What was the month and day of the first sale?
3. What was the month and day of the last sale?
4. Which animal type is most popular?
5. What was the total dollar amount made in the month of January?
6. What was the total dollar amount made in the month of June?
7. What is the average number of items bought in each transaction?
8. What are the top ten product titles by the number of sales?
9. What are the top ten product titles by total dollar amount made?
10. What is the proportion of returning customers?

### Question 1: Number of Orders

In [5]:
num_trans = df_cleaned.groupby('trans_id')['total_sales'].size().count()
num_trans

28022

There is a total of `28022` transactions made in the year 2021. A huge number for small business that just started. What else can we discover?

### Question 2: Alpa and Omega I

In [7]:
df_cleaned_arrangedbydate = df_cleaned.sort_values(by=['trans_year','trans_day'])
first_date = df_cleaned_arrangedbydate[['trans_month','trans_day']].iloc[0]
first_date = tuple(first_date)
print(first_date)

(1, 1)


The month of our first sale was `January` and the day was `01`. What an interesting way to kickstart the year don't you think?

### Question 3: Alpha and Omega II

In [8]:
df_cleaned_arrangedbydate = df_cleaned.sort_values(by=['trans_year','trans_month','trans_day'])
last_date = df_cleaned_arrangedbydate[['trans_month','trans_day']].iloc[-1]
last_date = tuple(last_date)
print(last_date)

(6, 30)


The month of our last sale was `June` and the day was `30`.  Now we know the date of the first and last sale. Though this does not really provide us any meaningful insight, but it helps us get some understanding of our customer base. We will dive into deeper exploratory analyis shortly.

### Question 4: Cats vs Dogs

In [11]:
most_pop_1 = df_cleaned['prod_animal_type'].value_counts()
most_pop = most_pop_1.keys()[0]
print(most_pop)

cat


We see that the most popular animal product type is `cat`. It would be good to understand why cat food is more popular in some geographical regions as opposed to others. 

### Question 5: More Money More Problems I

In [12]:
jan_rev = df_cleaned.groupby('trans_month')['total_sales'].sum().iloc[0]
jan_rev = float(jan_rev)
print(jan_rev)

51739.73999999996


We are getting into the heart of the matter - aggregrating sales revenue by month. We randomly picked the month of January and we see that the sales revenue amounted to `$51739.73`

### Question 6: More Money More Problems II

In [13]:
june_rev = df_cleaned.groupby('trans_month')['total_sales'].sum().iloc[5]
june_rev = float(june_rev)
print(june_rev)

548822.7299999722


Again, we randomly picked the month of June to discover the total amount in sales revenue which was `$548822.72`. We can quickly notice that there has been an increase (almost 1000% increase) in the sales revenue from January to June. Is that a pattern or it's seasonal? We shall come to find out.

### Question 7: Transaction Size

In [15]:
avg_num = df_cleaned.groupby('trans_id')['trans_quantity'].sum()
avg_list = list(avg_num)
y=0
for x in avg_list:
    y = y + int(x)
z = len(avg_list)
avg_num_items = y/z
print(avg_num_items)

1.8745628434801227


From our analysis we discover that on average clients make `1.8745628434801227` orders in a single transaction. For simplicity of our analysis, we can say on average clients make `2` orders in a single transaction.

### Question 8: Best Products I

In [16]:
df_topten = df_cleaned.groupby('prod_title')['trans_quantity'].sum().sort_values(ascending = False)[:10]
top_num_sales = list(df_topten.keys())
print(top_num_sales)

['Reddy Beddy', 'Yum Fish-Dish', 'Kitty Climber', 'Feline Fix Mix', 'Tuna Tasties', 'Chewie Dental', 'Purrfect Puree', 'Whole Chemistry Recipe', 'Cat Cave', 'Snoozer Hammock']


### Question 9: Best Products II

In [17]:
df_toptensales = df_cleaned.groupby('prod_title')['total_sales'].sum().sort_values(ascending = False)[:10]
top_tot_sales = list(df_toptensales.keys())
print(top_tot_sales)

['Reddy Beddy', 'Cat Cave', 'Kitty Climber', 'Snoozer Hammock', 'Snoozer Essentails', 'Yum Fish-Dish', 'Scratchy Post', 'Feline Fix Mix', 'Foozy Mouse', 'Tuna Tasties']


### Question 10: Loyalty

In [18]:
total_cust = df_cleaned['cust_id'].nunique()
count_return = list(df_cleaned.groupby('cust_id').size())
number_of_return_customers = 0
for x in count_return:
    if x > 1:
        number_of_return_customers = number_of_return_customers + 1
prop_returning = number_of_return_customers/total_cust
print(prop_returning)

0.4522385951697189


Interesting to see that almost `50%` of customers are repeat buyers. This is a very good observation for our business as we can seek to understand how to these repeat buyers inorder to keep them with targeted loyalty programs such as coupons and special offers. 

These 10 questions do not completely answer our business objectives but they give us a headstart and point us in the right direction as we dive deeper into our data through visualizations. What visuals can tell about our data is more than a word 1000 words.